programs/rewards/src/instructions/deposit_mining.rs

```diff
@@ -33,6 +36,22 @@ pub fn process_deposit_mining<'a>(
 
     let delegate_mining = get_delegate_mining(delegate_mining, mining)?;
 
+    if let Some(delegate_mining) = delegate_mining {
+        if *delegate_mining.key
+            != verify_mining_address(
+                program_id,
+                delegate_wallet_addr,
+                reward_pool.key,
+                WrappedMining::from_bytes_mut(&mut delegate_mining.data.borrow_mut())?
```

Comment:
This doesn't have to be a mutable. We have a very special WrappedImmutableMining for this